coordinator/src/main/java/org/apache/uniffle/coordinator/HealthSelectStorageStrategy.java

```diff
@@ -0,0 +1,272 @@
+    this.availableRemoteStorageInfo = Maps.newHashMap();
+    this.sizeList = Lists.newCopyOnWriteArrayList();
+    ScheduledExecutorService readWriteRankScheduler = Executors.newSingleThreadScheduledExecutor(
+        ThreadUtils.getThreadFactory("readWriteRankScheduler-%d"));
+    // should init later than the refreshRemoteStorage init
+    readWriteRankScheduler.scheduleAtFixedRate(this::checkReadAndWrite, 1000,
+        cf.getLong(CoordinatorConf.COORDINATOR_REMOTE_STORAGE_SCHEDULE_TIME), TimeUnit.MILLISECONDS);
+  }
+
+  public void checkReadAndWrite() {
+    try {
+      if (remoteStoragePathRankValue.size() > 1) {
+        for (String path : remoteStoragePathRankValue.keySet()) {
+          Path remotePath = new Path(path);
+          fs = HadoopFilesystemProvider.getFilesystem(remotePath, conf);
+          Path testPath = new Path(path + "/rssTest");
+          long startWriteTime = System.currentTimeMillis();
+          try {
+            byte[] data = RandomUtils.nextBytes(fileSize);
+
+            try (FSDataOutputStream fos = fs.create(testPath)) {
+              fos.write(data);
+              fos.flush();
+            }
+            byte[] readData = new byte[fileSize];
+            int readBytes;
+            try (FSDataInputStream fis = fs.open(testPath)) {
+              int hasReadBytes = 0;
+              do {
+                readBytes = fis.read(readData);
+                if (hasReadBytes < fileSize) {
+                  for (int i = 0; i < readBytes; i++) {
+                    if (data[hasReadBytes + i] != readData[i]) {
+                      RankValue rankValue = remoteStoragePathRankValue.get(path);
+                      remoteStoragePathRankValue.put(path, new RankValue(Long.MAX_VALUE, rankValue.getAppNum().get()));
+                    }
+                  }
+                }
+                hasReadBytes += readBytes;
+              } while (readBytes != -1);
+            }
+          } catch (Exception e) {
+            LOG.error("Storage read and write error ", e);
+            RankValue rankValue = remoteStoragePathRankValue.get(path);
+            remoteStoragePathRankValue.put(path, new RankValue(Long.MAX_VALUE, rankValue.getAppNum().get()));
+          } finally {
+            sortPathByIORank(path, testPath, startWriteTime);
+          }
+        }
+      } else {
+        sizeList = Lists.newCopyOnWriteArrayList(remoteStoragePathRankValue.entrySet());
+      }
+    } catch (Exception e) {
+      LOG.error("Some error happened, compare the number of apps to select a remote path", e);
+      sizeList = Lists.newCopyOnWriteArrayList(remoteStoragePathRankValue.entrySet()).stream().filter(Objects::nonNull)
+          .sorted(Comparator.comparingDouble(entry -> entry.getValue().getAppNum().get())).collect(Collectors.toList());
+    }
+  }
+
+  @VisibleForTesting
+  public void sortPathByIORank(String path, Path testPath, long startWrite) {
+    try {
+      fs.deleteOnExit(testPath);
```

Comment:
Why do use `deleteOnExit`?

Reply:
At first, I was worried that requesting NN too frequently would put pressure on NN, so I thought that the JVM process would be deleted only when it exited. But now I have tested it on the cluster. Using the `delete` operation, I can delete 100m of data every second. There is no pressure, so I will use the `delete` operation instead.